src/cephadm/cephadm

```diff
@@ -1873,11 +1873,16 @@ def infer_config(func: FuncT) -> FuncT:
             if not name:
                 daemon_list = list_daemons(ctx, detail=False)
                 for daemon in daemon_list:
-                    if daemon.get('name', '').startswith('mon.') and daemon.get('fsid', '') == ctx.fsid:
+                    if (
+                        daemon.get('name', '').startswith('mon.')
+                        and daemon.get('fsid', '') == ctx.fsid
+                        and daemon.get('style', '') == 'cephadm:v1'
+                        and os.path.exists(os.path.join(get_data_dir(ctx.fsid, ctx.data_dir, 'mon', daemon['name'].split('.', 1)[1]), 'config'))
```

Comment:
This os.path.join(...) line and the one below on line 1885 seem rather complex to read to me. The similarity between the two and complexity suggest to me that a small function may be in order.  It's not a requirement, but I think future readers would be able to comprehend it better.

Reply:
done